### ISSUE TITLE
 could not get source code

### ISSUE THREAD

**Issue by yishenggudou** [pyinstaller/pyinstaller#1945]
```
Failed to execute script cli
Traceback (most recent call last):
  File "cli.py", line 589, in <module>
  File "cli.py", line 214, in __init__
  File "cli.py", line 295, in buildProduct
  File "cli.py", line 314, in buildThirdPartyJs
  File "cli.py", line 525, in compress
  File "site-packages/slimit/minifier.py", line 37, in minify
  File "site-packages/slimit/parser.py", line 55, in __init__
  File "site-packages/slimit/lexer.py", line 98, in __init__
  File "site-packages/slimit/lexer.py", line 102, in build
  File "site-packages/ply/lex.py", line 908, in lex
  File "site-packages/ply/lex.py", line 580, in validate_all
  File "site-packages/ply/lex.py", line 823, in validate_rules
  File "site-packages/ply/lex.py", line 835, in validate_module
  File "inspect.py", line 690, in getsourcelines
  File "inspect.py", line 538, in findsource
IOError: could not get source code
```

**Comment by htgoebel** [comment]
You are providing much too less information to help you. For the case you are talking about the freezing you application (this is running PyInstaller), then obviously your library is not installed correctly. (But this is not a PyInstaller problem, since PyInstaller works without the source, do).

If this is a problem when running your frozen application: PyInstaller does not include any source files. If your application requires the source, you'll need to take care of this ourself. In your .spec-file you may collect the source for each file listed in `a.pure` and add it to `a.datas`. Not sure if this is all which is required to solve your issue.

**Comment by riggsd** [comment]
This issue (which also affected `py2app`) was fixed in PLY 3.9.

See: https://github.com/dabeaz/ply/issues/97

**Comment by htgoebel** [comment]
So I take this as "solved".
